jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/WebSocketConnection.java

```diff
@@ -394,7 +418,7 @@ public boolean moreDemand()
                 case NOT_DEMANDING ->
                 {
                     fillingAndParsing = false;
-                    if (!networkBuffer.hasRemaining())
+                    if (networkBuffer != null && !networkBuffer.hasRemaining())
```

Comment:
```suggestion
                    if (BufferUtil.hasContent(networkBuffer))
```

Reply:
Can't do because `networkBuffer` is a `RetainableByteBuffer`.

Unless you want me to make a new version of hasContent for this.